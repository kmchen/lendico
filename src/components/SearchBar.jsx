import React, { useState } from 'react';
import PropTypes from 'prop-types';
import TextField from '@material-ui/core/TextField';
import { makeStyles } from '@material-ui/core/styles';
import Container from '@material-ui/core/Container';
import Button from '@material-ui/core/Button';

const useStyles = makeStyles(theme => ({
  root: {
    flexGrow: 1,
  },
  textField: {
    marginRight: 20,
  },
  heroContent: {
    backgroundColor: '#f5f5f5',
    padding: theme.spacing(8, 0, 6),
  },
}));

function SearchBar({ durationChange, amountChange, getMonthlyInstallment }) {
  const [duration, setDuration] = useState(1);
  const [amount, setAmount] = useState(1);
  const [hasDurationError, setDurationError] = useState(false);
  const [hasAmountError, setAmountError] = useState(false);
  const handleDurationChange = event => {
    const { value: newDuration } = event.target;
    if (newDuration < 1) {
      setDurationError(true);
    } else {
      setDurationError(false);
    }
    setDuration(newDuration);
    durationChange(newDuration);
  };
  const handleAmountChange = event => {
    const { value: newAmount } = event.target;
    if (newAmount < 1) {
      setAmountError(true);
    } else {
      setAmountError(false);
    }
    setAmount(newAmount);
    amountChange(newAmount);
  };
  const classes = useStyles();
  return (
    <div className={classes.heroContent}>
      <Container maxWidth="sm">
        <div className={classes.root}>
          <TextField
            required
            id="standard-number"
            label={hasAmountError ? 'Amount has to be > 1' : 'Amount'}
            value={amount}
            onChange={handleAmountChange}
            type="number"
            className={classes.textField}
            InputLabelProps={{
              shrink: true,
            }}
            margin="normal"
            error={hasAmountError}
          />
          <TextField
            required
            id="standard-number"
            label={hasDurationError ? 'Duration has to be > 1' : 'Duration'}
            value={duration}
            onChange={handleDurationChange}
            type="number"
            className={classes.textField}
            InputLabelProps={{
              shrink: true,
            }}
            margin="normal"
            error={hasDurationError}
          />
          <Button
            disabled={hasAmountError || hasDurationError}
            variant="contained"
            component="span"
            className={classes.button}
            onClick={getMonthlyInstallment}
          >
            OK
          </Button>
        </div>
      </Container>
    </div>
  );
}

SearchBar.propTypes = {
  getMonthlyInstallment: PropTypes.func.isRequired,
  durationChange: PropTypes.func.isRequired,
  amountChange: PropTypes.func.isRequired,
};

export default SearchBar;
