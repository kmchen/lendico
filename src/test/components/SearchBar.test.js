import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import SearchBar from '../../components/SearchBar';

configure({ adapter: new Adapter() });

describe('SearchBar component', () => {
  let SearchBarComponent;

  beforeEach(() => {
    const props = {
      getMonthlyInstallment: jest.fn(),
      durationChange: jest.fn(),
      amountChange: jest.fn(),
    };
    SearchBarComponent = shallow(<SearchBar {...props} />);
  });

  it('should render SearchBar component', () => {
    expect(SearchBarComponent).toMatchSnapshot();
  });

  it('should render 2 textFields and 1 button', () => {
    const textFields = SearchBarComponent.find('WithStyles(ForwardRef(TextField))');
    const button = SearchBarComponent.find('WithStyles(ForwardRef(Button))');
    expect(textFields.exists()).toBeTruthy();
    expect(textFields.length).toEqual(2);
    expect(button.length).toEqual(1);
  });
});
