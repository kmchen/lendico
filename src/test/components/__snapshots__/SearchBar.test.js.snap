// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component should render SearchBar component 1`] = `
<div
  className="makeStyles-heroContent-3"
>
  <WithStyles(ForwardRef(Container))
    maxWidth="sm"
  >
    <div
      className="makeStyles-root-1"
    >
      <WithStyles(ForwardRef(TextField))
        InputLabelProps={
          Object {
            "shrink": true,
          }
        }
        className="makeStyles-textField-2"
        defaultValue="1"
        error={false}
        id="standard-number"
        label="Amount"
        margin="normal"
        onChange={[Function]}
        required={true}
        type="number"
        value={1}
      />
      <WithStyles(ForwardRef(TextField))
        InputLabelProps={
          Object {
            "shrink": true,
          }
        }
        className="makeStyles-textField-2"
        defaultValue="1"
        error={false}
        id="standard-number"
        label="Duration"
        margin="normal"
        onChange={[Function]}
        required={true}
        type="number"
        value={1}
      />
      <WithStyles(ForwardRef(Button))
        component="span"
        disabled={false}
        onClick={[MockFunction]}
        variant="contained"
      >
        OK
      </WithStyles(ForwardRef(Button))>
    </div>
  </WithStyles(ForwardRef(Container))>
</div>
`;
